zone/npc.cpp

```diff
@@ -1469,324 +1474,351 @@ NPC* NPC::SpawnNPC(const char* spawncommand, const glm::vec4& position, Client*
+
+	if (!s2.id) {
 		return 0;
 	}
-	spawnid = results.LastInsertedID();
 
-	query = StringFormat("INSERT INTO spawnentry (spawngroupID, npcID, chance) VALUES(%i, %i, %i)", last_insert_id,
-			     spawn->GetNPCTypeID(), 100);
-	results = QueryDatabase(query);
-	if (!results.Success()) {
+	auto se = SpawnentryRepository::NewEntity();
+
+	se.spawngroupID = sg.id;
+	se.npcID        = n->GetNPCTypeID();
+	se.chance       = 100;
+
+	if (!se.spawngroupID) {
 		return 0;
 	}
 
-	return spawnid;
+	return s2.id;
 }
 
-uint32 ZoneDatabase::UpdateNPCTypeAppearance(Client *client, NPC *spawn)
+uint32 ZoneDatabase::UpdateNPCTypeAppearance(Client* c, NPC* n)
 {
-	std::string query =
-	    StringFormat("UPDATE npc_types SET name = '%s', level = '%i', race = '%i', class = '%i', "
-			 "hp = '%i', gender = '%i', texture = '%i', helmtexture = '%i', size = '%i', "
-			 "loottable_id = '%i', merchant_id = '%i', face = '%i' "
-			 "WHERE id = '%i'",
-			 spawn->GetName(), spawn->GetLevel(), spawn->GetRace(), spawn->GetClass(), spawn->GetMaxHP(),
-			 spawn->GetGender(), spawn->GetTexture(), spawn->GetHelmTexture(), spawn->GetSize(),
-			 spawn->GetLoottableID(), spawn->MerchantType, spawn->GetLuclinFace(), spawn->GetNPCTypeID());
-	auto results = QueryDatabase(query);
-	return results.Success() == true ? 1 : 0;
-}
+	auto e = NpcTypesRepository::FindOne(*this, n->GetNPCTypeID());
 
-uint32 ZoneDatabase::DeleteSpawnLeaveInNPCTypeTable(const char *zone, Client *client, NPC *spawn)
-{
-	uint32 id = 0;
-	uint32 spawngroupID = 0;
+	e.name         = Strings::RemoveNumbers(n->GetName());
+	e.level        = n->GetLevel();
+	e.race         = n->GetRace();
+	e.class_       = n->GetClass();
+	e.hp           = n->GetMaxHP();
+	e.gender       = n->GetGender();
+	e.texture      = n->GetTexture();
+	e.helmtexture  = n->GetHelmTexture();
+	e.size         = n->GetSize();
+	e.loottable_id = n->GetLoottableID();
+	e.merchant_id  = n->MerchantType;
+	e.face         = n->GetLuclinFace();
 
-	std::string query = StringFormat("SELECT id, spawngroupID FROM spawn2 WHERE "
-					 "zone='%s' AND spawngroupID=%i",
-					 zone, spawn->GetSpawnGroupId());
-	auto results = QueryDatabase(query);
-	if (!results.Success())
-		return 0;
+	const int updated = NpcTypesRepository::UpdateOne(*this, e);
 
-	if (results.RowCount() == 0)
-		return 0;
+	return updated;
+}
 
-	auto row = results.begin();
-	if (row[0])
-		id = Strings::ToInt(row[0]);
+uint32 ZoneDatabase::DeleteSpawnLeaveInNPCTypeTable(const std::string& zone, Client* c, NPC* n)
+{
+	const auto& l = Spawn2Repository::GetWhere(
+		*this,
+		fmt::format(
+			"`zone` = '{}' AND `spawngroupID` = {}",
+			zone,
+			n->GetSpawnGroupId()
+		)
+	);
 
-	if (row[1])
-		spawngroupID = Strings::ToInt(row[1]);
+	if (l.empty()) {
+		return 0;
+	}
 
-	query = StringFormat("DELETE FROM spawn2 WHERE id = '%i'", id);
-	results = QueryDatabase(query);
-	if (!results.Success())
+	if (!Spawn2Repository::DeleteOne(*this, l[0].id)) {
```

Comment:
Let's use a different receiver other than `l[0]` for readability

```cpp
auto s = l[0];
```

Use s below